### PR TITLE
Dimmer implementation

### PR DESCRIPTION
**Pull request by MatthewWaanders**

### CODE REVIEW

File: drivers/FGD-211/device.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const ZwaveDevice = require('homey-meshdriver').ZwaveDevice;
+
+class FibaroDimmerDevice extends ZwaveDevice {
+	
+	onMeshInit() {
+        this._momentaryTrigger = new Homey.FlowCardTriggerDevice('FGD-211_momentary', this._switchTriggersRunListener.bind(this));
+        this._toggleTrigger = new Homey.FlowCardTriggerDevice('FGD-211_toggle', this._switchTriggersRunListener.bind(this));
+        this._rollerTrigger = new Homey.FlowCardTriggerDevice('FGD-211_roller', this._switchTriggersRunListener.bind(this));
+
+		this.registerCapability('onoff', 'SWITCH_MULTILEVEL');
```
Comment (RobinBol) [inline drivers/FGD-211/device.js]: Strange indentation

File: drivers/FGD-211/driver.compose.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      265,
+      4106,
+      8202,
+      12298
+    ],
+    "zwaveAllianceProductId": 88,
+    "learnmode": {
+      "image": "{{driverAssetsPath}}/learnmode.svg",
+      "instruction": {
+        "en": "Press the button on your Fibaro Dimmer three times",
```
Comment (RobinBol) [inline drivers/FGD-211/driver.compose.json]: {{driverNameEn}} please check other drivers as well

File: drivers/FGD-211/driver.flow.compose.json [hunk 3]
```diff
@@ -0,0 +1,271 @@
+{
+  "trigger": [
+    {
+      "id": "FGD-211_momentary",
+      "title": {
+        "en": "Dimmer scene (momentary)",
+        "nl": "Dimmer scene (pulsdruk)"
+      },
+      "hint": {
+        "en": "This card will only get triggered if the setting Switch type: \"Momentary switch\" is selected.",
```
Comment (RobinBol) [inline drivers/FGD-211/driver.flow.compose.json]: card -> Flow Trigger
Switch type -> \"Switch type\" or 'if the switch type setting...'

File: drivers/FGD-211/driver.flow.compose.json [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+        "nl": "Deze kaart wordt alleen geactiveerd als de instelling Schakelaar type: \"Pulsdruk schakelaar\" is geselecteerd."
```
Comment (RobinBol) [inline drivers/FGD-211/driver.flow.compose.json]: same here

File: drivers/FGD-211/driver.flow.compose.json [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+      ]
+    },
+    {
+      "id": "FGD-211_toggle",
+      "title": {
+        "en": "Dimmer scene (toggle)",
+        "nl": "Dimmer scene (tuimel)"
+      },
+      "hint": {
+        "en": "This card will only get triggered if the setting Switch type: \"Toggle Switch\" is selected. Extra setting options: * = Bistable switch: same direction, ** = Bistable switch: toggle output",
```
Comment (RobinBol) [inline drivers/FGD-211/driver.flow.compose.json]: card (see above)
switch type see above

File: drivers/FGD-211/driver.flow.compose.json [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+      ]
+    },
+    {
+      "id": "FGD-211_roller",
+      "title": {
+        "en": "Dimmer scene (roller Shutter)",
+        "nl": "Dimmer scene (rolgordijn)"
+      },
+      "hint": {
+        "en": "This card will only get triggered if the setting Switch type: \"Roller shutter switch\" is selected.",
```
Comment (RobinBol) [inline drivers/FGD-211/driver.flow.compose.json]: same

File: drivers/FGD-211/driver.settings.compose.json [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+        "id": "double_click",
+        "type": "checkbox",
+        "label": {
+          "en": "Double click",
+          "nl": "Dubbel klik"
+        },
+        "value": false,
+        "hint": {
+          "en": "This parameter ({{zwaveParameterIndex}}) determines whether or not double clicking sets the brightness level to maximum.",
+          "nl": "Deze parameter ({{zwaveParameterIndex}}) bepaalt of de helderheid naar maximaal geschakelt wordt bij dubbel klikken."
```
Comment (RobinBol) [inline drivers/FGD-211/driver.settings.compose.json]: 'schakelen' is not the right verb here

File: drivers/FGD-211/driver.settings.compose.json [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+          "index": 15,
+          "size": 1
+        }
+      },
+      {
+        "id": "3_way_switch",
+        "type": "checkbox",
+        "label": {
+          "en": "3-way switch function",
+          "nl": "Hotelschakeling Functie"
```
Comment (RobinBol) [inline drivers/FGD-211/driver.settings.compose.json]: check your capitalisation

File: drivers/FGD-211/driver.settings.compose.json [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+        },
+        "value": false,
+        "hint": {
+          "en": "This parameter ({{zwaveParameterIndex}}) determines whether switch no. 2 also controls the Dimmer 2 additionally.",
```
Comment (RobinBol) [inline drivers/FGD-211/driver.settings.compose.json]: for qubino I use consistent naming for the inputs/connected switches: 'Input 1/2/3' and 'Ingang 1/2/3' not sure if applicable here als long as it is consistent

File: drivers/FGD-211/driver.settings.compose.json [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+          "nl": "Deze parameter ({{zwaveParameterIndex}}) bepaalt of schakelaar 2 schakelt ook Dimmer 2 bediend."
```
Comment (RobinBol) [inline drivers/FGD-211/driver.settings.compose.json]: weird sentence
bedient*

File: drivers/FGD-211/driver.settings.compose.json [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+        "id": "switch_type",
+        "type": "dropdown",
+        "label": {
+          "en": "Switch type",
+          "nl": "Schakelaar type"
+        },
+        "value": "0",
+        "hint": {
+          "en": "This parameter ({{zwaveParameterIndex}}) determines whether this device behaves like a momentary switch or a toggle switch.",
+          "nl": "Deze parameter ({{zwaveParameterIndex}}) bepaalt of dit apparaat zich gedraag als een kortstondige schakelaar of een tuimelschakelaar."
```
Comment (RobinBol) [inline drivers/FGD-211/driver.settings.compose.json]: gedraagt*

File: drivers/FGD-211/driver.settings.compose.json [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+          "nl": "Stapgrootte dimmen bij automatische besturing"
+        },
+        "value": 1,
+        "attr": {
+          "min": 1,
+          "max": 99
+        },
+        "hint": {
+          "en": "This parameter ({{zwaveParameterIndex}}) determines the percentage of a dimming step at automatic control.",
+          "nl": "Deze parameter ({{zwaveParameterIndex}}) bepaalt Het percentageper dim stap bij automatische besturing."
```
Comment (RobinBol) [inline drivers/FGD-211/driver.settings.compose.json]: het*

Comment (RobinBol) [reply drivers/FGD-211/driver.settings.compose.json]: percentage per*

File: drivers/FGD-211/driver.settings.compose.json [hunk 13]
```diff
@@ -0,0 +1,240 @@
...
+          "nl": "Percentage van dim stap bij handmatige bediening"
+        },
+        "value": 1,
+        "attr": {
+          "min": 1,
+          "max": 99
+        },
+        "hint": {
+          "en": "This parameter ({{zwaveParameterIndex}}) determines the percentage of a dimming step at manual control.",
+          "nl": "Deze parameter ({{zwaveParameterIndex}}) bepaalt Het percentageper dim stap bij handmatige besturing."
```
Comment (RobinBol) [inline drivers/FGD-211/driver.settings.compose.json]: het
percentage per

File: drivers/FGD-211/device.js (hunk 1 above)
Comment (MatthewWaanders) [reply drivers/FGD-211/device.js]: Can confirm this is caused by my IDE. I'll figure out a way to fix it!

File: drivers/FGD-211/driver.flow.compose.json [hunk 14]
```diff
@@ -7,8 +7,8 @@
         "nl": "Dimmer scene (pulsdruk)"
       },
       "hint": {
-        "en": "This card will only get triggered if the setting Switch type: \"Momentary switch\" is selected.",
-        "nl": "Deze kaart wordt alleen geactiveerd als de instelling Schakelaar type: \"Pulsdruk schakelaar\" is geselecteerd."
+        "en": "This Flow Trigger will only get activated if the wtich type setting is set to \"Momentary switch\".",
```
Comment (RobinBol) [inline drivers/FGD-211/driver.flow.compose.json]: typo

File: drivers/FGD-211/driver.flow.compose.json [hunk 15]
```diff
@@ -7,8 +7,8 @@
...
+        "nl": "Deze Flow Trigger wordt alleen geactiveerd als de instelling schakelaar type \"Pulsdruk schakelaar\" is geselecteerd."
```
Comment (RobinBol) [inline drivers/FGD-211/driver.flow.compose.json]: 'als de instelling schakelaar type op \".."\ is ingesteld'*

File: drivers/FGD-211/driver.flow.compose.json [hunk 16]
```diff
@@ -96,8 +96,8 @@
         "nl": "Dimmer scene (tuimel)"
       },
       "hint": {
-        "en": "This card will only get triggered if the setting Switch type: \"Toggle Switch\" is selected. Extra setting options: * = Bistable switch: same direction, ** = Bistable switch: toggle output",
-        "nl": "Deze kaart wordt alleen geactiveerd als de instelling Schakelaar type: \"Tuimel Schakelaar\" is geselecteerd. Extra instelling opties: * = Bistabiele schakelaar: zelfde richting, ** = Bistabiele schakelaar: schakelt uitgang"
+        "en": "This Flow Trigger will only get activated if the switch type setting is set to \"Toggle Switch\". Extra setting options: * = Bistable switch: same direction, ** = Bistable switch: toggle output",
+        "nl": "Deze Flow Trigger wordt alleen geactiveerd als de instelling schakelaar type \"Tuimel Schakelaar\" is geselecteerd. Extra instelling opties: * = Bistabiele schakelaar: zelfde richting, ** = Bistabiele schakelaar: schakelt uitgang"
```
Comment (RobinBol) [inline drivers/FGD-211/driver.flow.compose.json]: see above

File: drivers/FGD-211/driver.flow.compose.json [hunk 17]
```diff
@@ -199,8 +199,8 @@
         "nl": "Dimmer scene (rolgordijn)"
       },
       "hint": {
-        "en": "This card will only get triggered if the setting Switch type: \"Roller shutter switch\" is selected.",
-        "nl": "Deze kaart wordt alleen geactiveerd als de instelling Schakelaar type: \"Rolgordijn schakelaar\" is geselecteerd."
+        "en": "This Flow Trigger will only get activated if the switch type setting is set to \"Roller shutter switch\".",
+        "nl": "Deze Flow Trigger wordt alleen geactiveerd als de instelling schakelaar type \"Rolgordijn schakelaar\" is geselecteerd."
```
Comment (RobinBol) [inline drivers/FGD-211/driver.flow.compose.json]: see above

File: drivers/FGD-211/driver.settings.compose.json [hunk 18]
```diff
@@ -17,7 +17,7 @@
         "value": false,
         "hint": {
           "en": "This parameter ({{zwaveParameterIndex}}) determines whether or not double clicking sets the brightness level to maximum.",
-          "nl": "Deze parameter ({{zwaveParameterIndex}}) bepaalt of de helderheid naar maximaal geschakelt wordt bij dubbel klikken."
+          "nl": "Deze parameter ({{zwaveParameterIndex}}) bepaalt of de helderheid naar maximaal wordt gezet bij tweemaal drukken."
```
Comment (RobinBol) [inline drivers/FGD-211/driver.settings.compose.json]: of de helderheid op maximaal wordt ingesteld*

**Pull request merged**